toolchain/lc-compile/src/main.c

```diff
@@ -115,9 +129,41 @@ static void full_main(int argc, char *argv[])
     for (argi = 0; argi < argc; ++argi)
     {
         const char *opt = argv[argi];
-        const char *optarg = (argi + 1 < argc) ? argv[argi+1] : NULL;
+        const char *optarg = (argi + 1 < argc) && 0 != strncmp(argv[argi+1], "--", 2) ? argv[argi+1] : NULL;
         if (!end_of_args)
         {
+            if (0 == strcmp(opt, "--order"))
```

Comment:
Since both `--order` and `--changed-order` are both covered by `--deps`, these options seem no longer necessary.